### DISCUSSION TITLE
How can I use detection.detect(file1.read()) to detect file in memory?

### DISCUSSION THREAD

**Discussion by martinenkoEduard** [exadel-inc/CompreFace#829] [Q&A] [answered]
How can I use detection.detect(file1.read()) to detect file in memory?

@app.route("/detectAndSaveAllFaces", methods=["POST"])
def upload_file():
    if request.method == "POST":
        if "file1" not in request.files:
            return "there is no file1 in form!"
        file1 = request.files["file1"]
        #text1 = request.form['text1']

        path = os.path.join(staticFolderImgPath, file1.filename)
        filemd5 = hashlib.md5(file1.read()).hexdigest() 
        filesha1 = hashlib.sha1(file1.read()).hexdigest()            
        response = detection.detect(file1)

I don't want to save this file.
Can I pass file bytes to detection.detect() function?

**Comment by martinenkoEduard** [comment]
It returns {'message': 'No face is found in the given image', 'code': 28}

**Comment by pospielov** [comment, marked as answer]
This method supports url, local path or bytes as an argument. 
You can look at how it is used in the demo, we also send bytes to this method
https://github.com/exadel-inc/compreface-python-sdk/blob/main/webcam_demo/compreface_webcam_detection_demo.py
